js/client/modules/@arangodb/arango-collection.js

```diff
@@ -535,15 +535,17 @@ ArangoCollection.prototype.truncate = function (options) {
   } else {
     options = options || {};
   }
-
+  if (!options.hasOwnProperty('compact')) {
+    options.compact = true;
+  }
   let headers = {};
   if (options && options.transactionId) {
     headers['x-arango-trx-id'] = options.transactionId;
   }
 
   var append = (options.waitForSync ? '&waitForSync=true' : '');
+  append += (append === '')? '?':'&' + (options.compact ? 'compact=true' : 'compact=false');
   var requestResult = this._database._connection.PUT(this._baseurl('truncate') + append, null, headers);
-
   arangosh.checkRequestResult(requestResult);
   // invalidate cache
   this._status = null;
```

Comment:
Below this comments, the function ends with
```
  if (!options.compact) {
    return;
  }
```
. That seems to be a relict, can you remove it?